### ISSUE TITLE
Same precision@k with different item embeddings?

### ISSUE THREAD

**Issue by BianchiGiulia** [lyst/lightfm#714]
Hello!

I am experimenting with various features with the aim to track how these impact the model performance. I use the [mind dataset](https://msnews.github.io/), a collection of news and user behaviour, plus many other meta-textual data.
I started with a normal pure collaborative filtering approach, where my data was had just columns    `user_id, item_id, label` with label being 1 or 0.

Then I added `category` as item feature, in a one-hot encoding fashion as weights, i.e.
```
[(item_0,
  {'category_0': weight_0,
   'category_1': weight_1,
   'category_2': weight_2,
   ...})
(item_1, ...
```
**n.b.: encoding without specifying the encoding in the weights did not seem to work.** In this way I found a slight improvement. 
So far so good, but when I tried to encode other kind of features e.g. subcategory, or tf-idf (of news title) in the same way, I get the same result at every epoch. I actually tried at least 5 different features.
I read in [another issue](https://github.com/lyst/lightfm/issues/486) that the final item embedding is the sum of the features embedding for a item, so I tried to use non-normalized tf-idf (so the sum wouldn't be always 1), but again, same result.

Hereunder I report the code to prepare data (full is my interaction data):
```
dataset = Dataset(item_identity_features=True) 
dataset.fit_partial(
    users = full['user_id'].unique(),
    items = full['item_id'].unique(),
    item_features = list(tfidf_data.columns)
)

(interactions_train, weights_train) = dataset.build_interactions([(x[1], x[2], x[3]) for x in train.itertuples()]) #x[0] is the row-index
(interactions_val, weights_test) = dataset.build_interactions([(x[1], x[2], x[3]) for x in val.itertuples()])

item_features = dataset.build_item_features(item_features_list, normalize=True)
[dataset_user_id_mapping,
 dataset_user_feature_mapping,
 dataset_item_id_mapping,
 dataset_item_feature_mapping] = dataset.mapping()

```
even using `normalize=False` doesn't change anything.
and the hyperparameters on which i tested the models:

```
learning_rates = [0.005, 0.01, 0.05]       
no_components = [20, 40, 80, 100]
loss_functions = ['bpr', 'warp']
max_sampled = 3

epochs = 10
patience = 3 #early stopping
k = 5
 ```

I evaluate on **precision@k**, which sadly is constant in all my experiment with a hybrid model. 
Any advice on what I might be doing wrong is much appreciated (and needed), 

thank you in advance!
